src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -738,10 +738,10 @@ static void ensure_join(JavaThread* thread) {
   ObjectLocker lock(threadObj, thread);
   // Thread is exiting. So set thread_status field in  java.lang.Thread class to TERMINATED.
   java_lang_Thread::set_thread_status(threadObj(), JavaThreadStatus::TERMINATED);
-  // Clear the native thread instance - this makes isAlive return false and allows the join()
-  // to complete once we've done the notify_all below. Needs a release() to obey Java Memory Model
-  // requirements.
-  OrderAccess::release();
+  // Clear the native thread instance - this makes isAlive return false and
+  // allows the join() to complete once we've done the notify_all() below.
+  // Needs a release() to obey Java Memory Model requirements (which is done
+  // in set_thread()).
```

Comment:
With the suggested renaming you don't need to add comments like this.

Reply:
Most of the comment isn't mine; I just updated it slightly and reformatted it. Is the
mention of "Java Memory Model requirements" not a useful thing to note here?

Reply:
Sorry I only meant your addition to the comment. The rest is important, if somewhat vague. The requirements is that if you call `t.isAlive()` and get false (because it has terminated) then you must see all stores performed by `t` before it terminated. We in part achieve this by marking `eetop` as a volatile field in the j.l.Thread object, hence the VM (technically) should always do a "volatile write" to it (i.e. release semantics). Conversely, reads of `eetop` from within the VM should have volatile-read semantics (i.e. acquire).